src/BlendingEditor/Website/ViewModels/CompositionPage.json.cs

```
@@ -36,7 +36,44 @@ public string FilePath
         {
             get
             {
-                var path = this.Data != null ? FileSystemHelper.GetAbsoluteHtmlFilePath(this.Data) : null;
+                var path = this.Data != null && FileSystemHelper.IsRegisteredCompositionsFolder ? HtmlViewCompositionHelper.GetHtmlFileName(this.Data) : null;
+                return path;
+            }
+            set
+            {
+                var newFileName = value;
+
+                if (newFileName == "") //this means "restore the initial file name"
+                {
+                    newFileName = HtmlViewCompositionHelper.GetInitialHtmlFileName(this.Data);
+                }
+
+                if (!FileSystemHelper.IsValidFileName(newFileName))
+                {
+                    return;
+                }
+
+                try
+                {
+                    string newPath = FileSystemHelper.FilePathToAbsoluteFilePath(newFileName);
+                    FileSystemHelper.MoveFileOrDeleteIfExists(this.Data, newPath);
+                    Db.Transact(() =>
+                    {
+                        this.Data.HtmlFilePath = value;
+                    });
+                }
+                catch (Exception)
```

Are you sure you want to just silently catch exception here without doing anything? I would suggest showing here some error message.